packages/react-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -16,6 +17,12 @@ export interface NumberInputProps extends React.HTMLProps<HTMLDivElement> {
   widthChars?: number;
   /** Indicates the whole number input should be disabled */
   isDisabled?: boolean;
+  /** Value to indicate if the input is modified to show that validation state.
+   * If set to warning, input will be modified to indicate warning state.
+   * If set to success, input will be modified to indicate valid state.
+   * If set to error,  input will be modified to indicate error state.
+   */
```

Comment:
```suggestion
  /** Value to indicate the input's validation state when the input is modified. */
```
Not sure if we need the rest of the test i.e. "If set to...", but not a blocker if we keep it in.